presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -198,10 +204,88 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         CassandraTableHandle handle = checkType(table, CassandraTableHandle.class, "table");
         CassandraPartitionResult result = partitionManager.getPartitions(handle, constraint.getSummary());
-        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions()));
+        ConnectorTableLayout layout = getTableLayout(session, new CassandraTableLayoutHandle(handle, result.getPartitions(), buildClusteringPredicate(schemaProvider.getTable(handle), constraint)));
         return ImmutableList.of(new ConnectorTableLayoutResult(layout, result.getUnenforcedConstraint()));
     }
 
+    private static Optional<String> buildClusteringPredicate(CassandraTable cassandraTable, Constraint<ColumnHandle> constraint)
+    {
+        List<CassandraColumnHandle> clusteringKeyColumns = cassandraTable.getClusteringKeyColumns();
+        Set<List<Object>> clusteringKeysSet = getClusteringKeysSet(cassandraTable, constraint.getSummary());
+        if (clusteringKeysSet == null || clusteringKeysSet.isEmpty()) {
```

Comment:
`getClusteringKeysSet`  doesn't return `null`. Just check whether it is not empty.